pytorch_lightning/utilities/enums.py

```diff
@@ -256,3 +257,22 @@ def interactive_compatible_types() -> List["_StrategyType"]:
     def is_interactive_compatible(self) -> bool:
         """Returns whether self is interactive compatible."""
         return self in _StrategyType.interactive_compatible_types()
+
+
+class _AcceleratorType(LightningEnum):
+    """Define Accelerator type by its nature.
+
+    >>> _AcceleratorType.CPU == _AcceleratorType.from_str('cpu')
+    True
+    >>> # you can match the type with string
+    >>> _AcceleratorType.GPU == 'GPU'
+    True
+    >>> # which is case invariant
+    >>> _AcceleratorType.TPU in ('tpu', 'CPU')
+    True
+    """
+
+    CPU = "CPU"
+    GPU = "GPU"
```

Comment:
Just a question: Should we also change `GPU` to `CUDA` to also allow for ROCm devices in the future?

Reply:
Then user would have to pass `accelerator="cuda"`, are we okay with that change?

Reply:
This would be an additional breaking change, but definitely cleaner and more aligned with PyTorch API. @ananthsub.

I believe it might be worth waiting for adding support `ROCm` as the GPUAccelerator might be renamed CUDAAccelerator too.

Reply:
We could also add it now and deprecate the gpu thing in the meantime. Fine with both ways :)

Reply:
I would like to avoid adding things prematurely which are not used yet. Avoids issues like #9956